pyqtgraph/graphicsItems/InfiniteLine.py

```diff
@@ -37,79 +78,125 @@ def __init__(self, pos=None, angle=90, pen=None, movable=False, bounds=None):
+        self.anchorUp = (0.5, 1.)
+        self.anchorDown = (0.5, 0.)
+        self.text = TextItem(fill=textFill)
+        self.text.setParentItem(self) # important
         self.p = [0, 0]
-        self.setAngle(angle)
-        if pos is None:
-            pos = Point(0,0)
-        self.setPos(pos)
 
         if pen is None:
             pen = (200, 200, 100)
-        
+
         self.setPen(pen)
-        self.setHoverPen(color=(255,0,0), width=self.pen.width())
+
+        if hoverPen is None:
+            self.setHoverPen(color=(255,0,0), width=self.pen.width())
+        else:
+            self.setHoverPen(hoverPen)
         self.currentPen = self.pen
-      
+
+        self.setMovable(movable)
+
+        if pos is None:
+            pos = Point(0,0)
+        self.setPos(pos)
+
+        if (self.angle == 0 or self.angle == 90) and self.label:
+            self.text.show()
+        else:
+            self.text.hide()
+
+
     def setMovable(self, m):
         """Set whether the line is movable by the user."""
         self.movable = m
         self.setAcceptHoverEvents(m)
-      
+
     def setBounds(self, bounds):
         """Set the (minimum, maximum) allowable values when dragging."""
         self.maxRange = bounds
         self.setValue(self.value())
-        
+
     def setPen(self, *args, **kwargs):
-        """Set the pen for drawing the line. Allowable arguments are any that are valid 
+        """Set the pen for drawing the line. Allowable arguments are any that are valid
         for :func:`mkPen <pyqtgraph.mkPen>`."""
         self.pen = fn.mkPen(*args, **kwargs)
         if not self.mouseHovering:
             self.currentPen = self.pen
             self.update()
-        
+
     def setHoverPen(self, *args, **kwargs):
-        """Set the pen for drawing the line while the mouse hovers over it. 
-        Allowable arguments are any that are valid 
+        """Set the pen for drawing the line while the mouse hovers over it.
+        Allowable arguments are any that are valid
         for :func:`mkPen <pyqtgraph.mkPen>`.
-        
+
         If the line is not movable, then hovering is also disabled.
-        
+
         Added in version 0.9.9."""
         self.hoverPen = fn.mkPen(*args, **kwargs)
         if self.mouseHovering:
             self.currentPen = self.hoverPen
             self.update()
-        
+
     def setAngle(self, angle):
         """
         Takes angle argument in degrees.
         0 is horizontal; 90 is vertical.
-        
-        Note that the use of value() and setValue() changes if the line is 
+
+        Note that the use of value() and setValue() changes if the line is
         not vertical or horizontal.
         """
         self.angle = ((angle+45) % 180) - 45   ##  -45 <= angle < 135
-        self.resetTransform()
-        self.rotate(self.angle)
+        # self.resetTransform()   # no longer needed since version 0.9.11
+        # self.rotate(self.angle) # no longer needed since version 0.9.11
```

Comment:
I should also emphasize that the rotation inheritance is something I did not want in this case because the label is a TextItem whose parent is self. If I apply a transformation, as it was done before, that transformation is also applied to the label. This behaviour results in something ugly (label rotated and scaled, not really readable actually). These API changes were a necessity considering the modification I wanted to add to this object (no such major changes will be necessary for the other modification I plan to propose).

I did not find another way (and I do not know if there is another way).